lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -60,6 +60,22 @@ impl ShardReplicaSet {
         Ok(())
     }
 
+    pub fn take_partial_snapshot_recovery_lock(
+        &self,
+    ) -> CollectionResult<tokio::sync::OwnedMutexGuard<()>> {
+        self.partial_snapshot_meta.take_recovery_lock()
+    }
+
+    pub fn take_partial_snapshot_read_lock(
+        &self,
+    ) -> CollectionResult<tokio::sync::OwnedSemaphorePermit> {
+        self.partial_snapshot_meta.take_read_lock()
+    }
+
+    pub fn check_partial_snapshot_read_lock(&self) -> CollectionResult<()> {
```

Comment:
Does it have to be partial-snapshot specific, or we can do it for any kind of snapshots?

Reply:
I think we can easily reuse it for all snapshot types, I just make it partial snapshot specific so far